ili/utils/ndes_pt.py

```diff
@@ -122,6 +122,8 @@ def forward(
         log_abs_det_jacobian = self.theta_transform.log_abs_det_jacobian(
             theta, theta  # just for shape
         )  # for Affine/IdentityTransform, this outputs a constant
+        if len(log_abs_det_jacobian.shape) > 1:
+            log_abs_det_jacobian = log_abs_det_jacobian.sum(dim=1)
```

Comment:
Please add a comment explaining why this could happen. And please can you add a unit test for this?

Reply:
This only seems to occur when `theta_normalize=False` and `theta_transform` is an `identity_transform`. In that case, `log_abs_det_jacobian=0` anyway. I don't know why its different, but the construction seems to be different in Pytorch: https://github.com/pytorch/pytorch/blob/5c2584a14c2283514703a17cba0a57c8bfb0d977/torch/distributions/transforms.py#L363

I've added a comment about this inline and added a case in the tests wherein `theta_normalize=False`, to cover this line